src/transform/transform.ts

```diff
@@ -564,43 +563,51 @@ function createResponseType(codeModel: CodeModel, group: OperationGroup, op: Ope
+        let prop = newProperty('PollUntilDone',
+          'PollUntilDone will poll the service endpoint until a terminal state is reached or an error is received',
+          newObject(`func(ctx context.Context, frequency time.Duration) (*${response.schema.language.go!.responseType.name}, error)`, 'TODO'));
         prop.schema.language.go!.lroPointerException = true;
-        (<Array<Property>>response.schema.language.go!.properties).push(prop);
+        (<Array<Property>>respTypeObject.language.go!.properties).push(prop);
         prop = newProperty('Poller', 'Poller contains an initialized poller', newObject(`${response.schema.language.go!.responseType.value}Poller`, 'TODO'));
         prop.schema.language.go!.lroPointerException = true;
-        (<Array<Property>>response.schema.language.go!.properties).push(prop);
+        (<Array<Property>>respTypeObject.language.go!.properties).push(prop);
+        // add the LRO response schema to the global list of response
+        const responseSchemas = <Array<Schema>>codeModel.language.go!.responseSchemas;
+        responseSchemas.push(respTypeObject);
+        response.schema.language.go!.lroResponseType = respTypeObject;
+      } else {
+        // exclude LRO headers from Widget response envelopes
+        // add any headers to the response type
+        for (const item of items(headers)) {
+          const prop = newProperty(item.key, item.value.description, item.value.schema);
+          prop.language.go!.fromHeader = item.value.header;
+          (<Array<Property>>response.schema.language.go!.properties).push(prop);
+        }
       }
+      // the Widget response doesn't belong in the poller response envelope
+      (<Array<Property>>response.schema.language.go!.properties).push(newProperty(propName, response.schema.language.go!.description, response.schema));
       if (!responseExists(codeModel, response.schema.language.go!.name)) {
         // add this response schema to the global list of response
         const responseSchemas = <Array<Schema>>codeModel.language.go!.responseSchemas;
         responseSchemas.push(response.schema);
-      } else if (isLROOperation(op)) {
-        // add this response schema with LRO fields to the global list of responses by 
-        // replacing the previously added response with the same name
-        const responseSchemas = <Array<Schema>>codeModel.language.go!.responseSchemas;
-        for (let i = 0; i < responseSchemas.length; i++) {
-          if (responseSchemas[i].language.go!.name === response.schema.language.go!.name) {
-            responseSchemas.splice(i, 1, response.schema);
-          }
-        }
       }
     }
     // create pageable type info
     if (isPageableOperation(op)) {
       if (codeModel.language.go!.pageableTypes === undefined) {
         codeModel.language.go!.pageableTypes = new Array<PagerInfo>();
       }
-      const name = `${(<SchemaResponse>response).schema.language.go!.name}Pager`;
+      let suffix = 'Pager';
```

Comment:
Should be `const`, or you could just do `const name = <name>+'Pager;'`.

Reply:
Done